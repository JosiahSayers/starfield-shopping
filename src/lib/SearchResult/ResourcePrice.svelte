<script lang="ts">
  import credit from "../../assets/credit.png";

  export let price: number;
</script>

<span class="badge badge-neutral m-1"
  ><img src={credit} alt="starfield credit logo" class="h-4" />{price}</span
>
