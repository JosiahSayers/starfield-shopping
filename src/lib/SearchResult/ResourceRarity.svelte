<script lang="ts">
  export let rarity: string;
  $: colorClass = ["rare", "unique", "uncommon", "exotic"].includes(
    rarity.toLowerCase()
  )
    ? rarity.toLowerCase()
    : "badge-neutral";
</script>

<span class={`badge ${colorClass} m-1`}>{rarity}</span>

<style>
  .rare {
    background-color: #a179d1;
    border-color: #a179d1;
    color: white;
  }

  .unique {
    background-color: #6fb374;
    border-color: #6fb374;
    color: white;
  }

  .uncommon {
    background-color: #6db5c5;
    border-color: #6db5c5;
    color: white;
  }

  .exotic {
    background-color: #cbb784;
    border-color: #cbb784;
    color: white;
  }
</style>
