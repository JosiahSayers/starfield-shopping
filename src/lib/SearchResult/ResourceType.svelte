<script lang="ts">
  export let type: string;
</script>

<span class="badge badge-neutral m-1">{type}</span>
