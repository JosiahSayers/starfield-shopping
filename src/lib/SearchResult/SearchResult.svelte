<script lang="ts">
  import ResourceBadges from "../ResourceBadges.svelte";
  import type { Resource } from "../stores/cart";

  export let resource: Resource;
</script>

<div class="flex flex-col">
  <span class="mb-1">
    {resource.name}
    {#if resource.shortName}
      - {resource.shortName}
    {/if}
  </span>
  <ResourceBadges {resource} />
</div>
