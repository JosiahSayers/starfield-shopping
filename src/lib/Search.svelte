<script lang="ts">
  import resources from "../data/resources.json";
  import SearchResult from "./SearchResult/SearchResult.svelte";
  import { cart, type Resource } from "./stores/cart";

  let wrapperRef: HTMLDivElement;
  let open = false;
  let searchValue = "";

  $: {
    open = searchValue.length > 0;
  }

  $: filteredList = resources.filter((resource) =>
    searchValue.length
      ? resource.name.toLowerCase().includes(searchValue.toLowerCase()) ||
        resource.shortName.toLowerCase().includes(searchValue.toLowerCase())
      : true
  );

  function handleSelect(resource: Resource) {
    cart.add({ ...resource, quantity: 1 });
    searchValue = "";
    (document.activeElement as any)?.blur?.();
  }
</script>

<div class="dropdown w-full" class:dropdown-open={open} bind:this={wrapperRef}>
  <input
    type="text"
    class="input input-bordered w-full"
    placeholder="Add Item..."
    bind:value={searchValue}
  />

  <div
    class="dropdown-content bg-base-200 top-14 max-h-96 overflow-auto flex-col rounded-md z-10"
  >
    <ul class="menu menu-compact" style={`width: ${wrapperRef?.clientWidth}px`}>
      {#each filteredList as resource}
        <li class="border-b border-b-base-content/10 w-full">
          <button on:click={() => handleSelect(resource)}
            ><SearchResult {resource} /></button
          >
        </li>
      {/each}
    </ul>
  </div>
</div>
