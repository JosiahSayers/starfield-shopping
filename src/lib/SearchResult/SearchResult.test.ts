import { describe, afterEach, it, expect } from 'vitest';
import { cleanup, render } from '@testing-library/svelte'
import SearchResult from './SearchResult.svelte'
import type { Resource } from '../stores/resources';
import Search from '../Search.svelte';

describe('SearchResult', () => {
  afterEach(() => cleanup());

  const testResource: Resource = {
    id: 1,
    name: 'Resource',
    shortName: 'Re',
    rarity: 'exotic',
    type: 'manufactured',
    mass: 1,
    value: 24,
    valueToMass: 24
  };

  it('mounts', () => {
    const { container } = render(SearchResult, { resource: testResource });
    expect(container).toBeTruthy();
    expect(container.innerHTML).toMatchSnapshot();
  });

  it('renders the name', () => {
    const { container } = render(SearchResult, { resource: testResource });
    expect(container.querySelector('span')?.textContent).toBe('Resource - Re');
  });

  it('renders the name when there is no short name', () => {
    const { container } = render(SearchResult, { resource: { ...testResource, shortName: '' } });
    expect(container.querySelector('span')?.textContent).toBe('Resource ');
  });
});
