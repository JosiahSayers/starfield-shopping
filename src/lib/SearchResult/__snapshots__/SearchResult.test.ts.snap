// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchResult > mounts 1`] = `"<div><div class=\\"flex flex-col\\"><span class=\\"mb-1\\">Resource - Re</span> <div class=\\"flex flex-wrap\\"><span class=\\"badge exotic m-1 svelte-1x1khpc\\">exotic</span> <span class=\\"badge badge-neutral m-1\\">manufactured</span> <span class=\\"badge badge-neutral m-1\\"><img src=\\"/src/assets/credit.png\\" alt=\\"starfield credit logo\\" class=\\"h-4\\">24</span></div></div></div>"`;
